trieste/models/gpflow/models.py

```diff
@@ -147,6 +150,16 @@ def _ensure_variable_model_data(self) -> None:
             ),
         )
 
+    def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        return self._posterior.predict_f(query_points)
+
+    def predict_joint(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        return self._posterior.predict_f(query_points, full_cov=True)
+
+    def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        f_mean, f_var = self.predict(query_points)
+        return self.model.likelihood.predict_mean_and_var(f_mean, f_var)
+
```

Comment:
can we move this to GPflowPredictor, of course if all models are using the same

Reply:
Not all the models do, but still a good suggestion. Done.

Reply:
Actually turns out that we need to initialise the posteriors at the end of initialisation so can't delegate it to the parent class, which makes refactoring a bit trickier. Might leave as is for now and take a look in a separate PR.

Reply:
I've had a second go at refactoring it which hopefully won't break all the tests!